### PR TITLE
samples: nrf_desktop: MCUBoot integration

### PR DESCRIPTION
**Pull request by pdunaj**

### CODE REVIEW

File: samples/nrf_desktop/CMakeLists.txt [hunk 1]
```diff
@@ -71,3 +74,27 @@ add_subdirectory(src/events)
 add_subdirectory(src/hw_interface)
 add_subdirectory(src/modules)
 add_subdirectory(src/services)
+
```
Comment (pdunaj) [inline samples/nrf_desktop/CMakeLists.txt]: below will add a command to sign hex file if you use
`ninja sign`

@SebastianBoe  is there a plan to add something like that in zephyr?

File: samples/nrf_desktop/CMakeLists.txt [hunk 2]
```diff
@@ -71,3 +74,27 @@ add_subdirectory(src/events)
...
+  if (NOT EXISTS ${MCUBOOT_DIR})
+    message(FATAL_ERROR "Path to MCUBoot must be specified")
+  endif()
+  set(cmd ${CMAKE_COMMAND} -E env
+      ${MCUBOOT_DIR}/scripts/imgtool.py sign
+      --key ${MCUBOOT_DIR}/root-rsa-2048.pem
+      --header-size ${CONFIG_TEXT_SECTION_OFFSET}
+      --align 8
+      --version ${CMAKE_PROJECT_VERSION}
+      --slot-size 0x69000
```
Comment (pdunaj) [inline samples/nrf_desktop/CMakeLists.txt]: slot-size will probably not be automated now - it is available from dts and parsed to config but is not visible in cmake for some reason

**Comment by pdunaj** [comment]
tested with PCA20041; application is built and booted via mcuboot

I am not adding command to flash signed image as at the moment it won't be helpful and in the long run maybe something will be implemented in zephyr.

File: samples/nrf_desktop/CMakeLists.txt [hunk 3]
```diff
@@ -71,3 +74,27 @@ add_subdirectory(src/events)
...
 add_subdirectory(src/modules)
 add_subdirectory(src/services)
+
+if (${CONFIG_BOOTLOADER_MCUBOOT})
+  if (NOT EXISTS ${MCUBOOT_DIR})
+    message(FATAL_ERROR "Path to MCUBoot must be specified")
+  endif()
+  set(cmd ${CMAKE_COMMAND} -E env
+      ${MCUBOOT_DIR}/scripts/imgtool.py sign
+      --key ${MCUBOOT_DIR}/root-rsa-2048.pem
```
[the remark this answers is not in the document]
Comment (pdunaj) [reply samples/nrf_desktop/CMakeLists.txt]: This config option is part of the mcuboot repo that we don't see in nrf one (i.e. it's a different project).
We could create an option to point to the the key... if not set we could use the default one from mcuboot and also emit a massage to make sure that default key should be used (as it is the same for everybody).

File: samples/nrf_desktop/CMakeLists.txt (hunk 2 above)
Comment (pdunaj) [reply samples/nrf_desktop/CMakeLists.txt]: I have found an option that is visible to cmake

File: samples/nrf_desktop/CMakeLists.txt (hunk 3 above)
[the remark this answers is not in the document]
Comment (pdunaj) [reply samples/nrf_desktop/CMakeLists.txt]: I have added a project option to specify path to the key. If default key is used warning is emitted.

File: samples/nrf_desktop/configuration/Kconfig [hunk 4]
```diff
@@ -0,0 +1,26 @@
+#
+# Copyright (c) 2018 Nordic Semiconductor
```
Comment (Qbicz) [inline samples/nrf_desktop/configuration/Kconfig]: 2019 ;)

**Comment by pdunaj** [comment]
@lemrey since you have seen the file already maybe you can give me a green tick as I need one :)

File: samples/nrf_desktop/CMakeLists.txt (hunk 1 above)
Comment (SebastianBoe) [reply samples/nrf_desktop/CMakeLists.txt]: Yes.

This is on it's way; https://github.com/zephyrproject-rtos/zephyr/commit/6679ff7647c29a553f835ac16b9680c0fd341be3#diff-93ebc7c448cc5ffd4336cc014cc05574R18

And also, another approach; https://github.com/zephyrproject-rtos/zephyr/pull/12903

**Pull request merged**